lib/request.coffee

```diff
@@ -83,11 +83,11 @@ module.exports = getRequest = ({
 					code: 'ResinRequestError'
 					statusCode: 401
 				, ->
-					return token.get().tap(token.remove).then (sessionToken) ->
+					return token.getKey().tap(token.removeKey).then (sessionToken) ->
```

Comment:
Also, take care because `ResinAuth` is a class, not a "static" module like resin-token, so if you separate the method from its instance, it will break

Reply:
@pimterry ```getKey``` is needed for the sessionToken, isn't it?

@flesler did you mean the ```token.removeKey``` that I will lose the context?

Reply:
> getKey is needed for the sessionToken, isn't it?

Ah, you're right - I'd just seen that we don't seem to need it for `remove` anymore, but you're totally correct. Good catch!

> did you mean the token.removeKey that I will lose the context?

I think you will, yes - `removeKey` uses `this` internally, so you need to make sure that still refers to the resin-auth instance when it gets called.

Reply:
Correct me if I'm wrong, but I think that this should not be a problem since all public "methods" in resin-auth are already binded to their context since they are using fat arrows to declare function expressions on properties instead of actual "methods". So the generated build uses `_this` for all these methods.
The only exception seems to be the two static methods of the JWT class (`parse` & `isValid`).

Keep in mind that I'm not against being extra cautious on loosing context on method-like calls.

See: https://github.com/resin-io-modules/resin-auth/blob/409eeec3e5db6783786cf00b067d59e54852680b/lib/auth.ts#L95

Reply:
@thgreasi great point! Yes, in that case @MoranF you can ignore the `this` part here, that should be fine.